src/sentry/static/sentry/app/components/asyncComponent.tsx

```diff
@@ -114,8 +112,8 @@ export default class AsyncComponent<
     this._measurement = {
       hasMeasured: false,
     };
-    if (props.router && props.router.routes) {
-      metric.mark(`async-component-${getRouteStringFromRoutes(props.router.routes)}`);
+    if (props.routes && props.routes) {
```

Comment:
`routes` should come from `props` instead of `props.router` based on the type. I am not sure why `props.router.routes` works because the field does not exist on the type. Maybe it was a deprecated field that you shouldn't use but we did from an earlier version of React Router?